### PR TITLE
Communication Console Tweak

### PR DESCRIPTION
**Pull request by Super3222**
#27

You can now log into communication consoles without needing to hold your ID in your hand.

### CODE REVIEW

**Comment by Groudonmaster** [comment]
:+1:

**Comment by FluffySurvivor** [comment]
:+1:
ty m8

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -63,20 +63,16 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
...
-					auth_id = "[I.registered_name] ([I.assignment])"
-					if((20 in I.access))
-						authenticated = 2
-				if(src.emagged)
+			var/mob/living/carbon/human/M = usr
+			var/obj/item/weapon/card/id/I = M.get_idcard()
+			if(src.allowed(M))
+				authenticated = 1
+				auth_id = "[I.registered_name] ([I.assignment])"
+				if((20 in I.access))
```
Comment (oisin100) [inline code/game/machinery/computer/communications.dm:71]: Will this not break if the user doesn't have a ID card? 

Other than that :+1:

Comment (Super3222) [inline code/game/machinery/computer/communications.dm:71]: @oisin100, I tested it and nothing broke.

auth_id is originally Unknown, and unless it's emagged it won't log you in without an ID.

Comment (oisin100) [inline code/game/machinery/computer/communications.dm:71]: Kk

File: code/game/machinery/computer/communications.dm [hunk 2]
```diff
@@ -63,20 +63,16 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
...
-				var/obj/item/device/pda/pda = I
-				I = pda.id
-			if (I && istype(I))
-				if(src.check_access(I))
-					authenticated = 1
-					auth_id = "[I.registered_name] ([I.assignment])"
-					if((20 in I.access))
-						authenticated = 2
-				if(src.emagged)
+			var/mob/living/carbon/human/M = usr
```
Comment (DavidEdwards) [inline code/game/machinery/computer/communications.dm:66]: Could be a problem - what if a silicon attempts to access the console?

Comment (Super3222) [inline code/game/machinery/computer/communications.dm:66]: @DavidEdwards 
Hm, that is a good point. I was likely to believe that line 68 would take care of that. Though it is worth testing. I'll get back to you on it.

Comment (Super3222) [inline code/game/machinery/computer/communications.dm:66]: Alright, I tested it as both silicon jobs (AI and cyborg). I was able to access the communication console normally.
All silicons have a universal access to all machines / computers in the code.

**Pull request merged**